src/components/Toolbar/index.jsx

```diff
@@ -0,0 +1,121 @@
+/* global cozy */
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import Menu, { Item } from 'components/Menu'
+import styles from './index.styl'
+
+import { withBreakpoints } from 'cozy-ui/react'
+
+const { BarRight } = cozy.bar
+
+class Toolbar extends Component {
+  render() {
+    const {
+      children,
+      moreMenuTitle,
+      moreMenuDisabled,
+      moreMenuClassName,
+      moreMenuButton,
+      breakpoints: { isMobile }
+    } = this.props
+
+    const actions = React.Children.toArray(children)
+    const highlightedActions = actions.filter(child => child.props.highlighted)
+    const secondaryActions = actions.filter(child => !child.props.highlighted)
+
+    const MoreMenu = (
+      <Menu
+        title={moreMenuTitle}
+        disabled={moreMenuDisabled}
+        className={moreMenuClassName}
+        button={moreMenuButton}
+      >
+        {highlightedActions.map(component =>
+          this.wrapInItem(
+            React.cloneElement(component, {
+              highlighted: false,
+              isSecondaryAction: false
+            })
+          )
+        )}
+        {secondaryActions.map(component =>
+          this.wrapInItem(
+            React.cloneElement(component, {
+              highlighted: false,
+              isSecondaryAction: true
+            })
+          )
+        )}
+      </Menu>
+    )
+
+    return (
+      <div role="toolbar">
+        {highlightedActions.map(component =>
+          React.cloneElement(component, {
+            highlighted: true,
+            isSecondaryAction: false
+          })
+        )}
+        {isMobile ? <BarRight>{MoreMenu}</BarRight> : MoreMenu}
+      </div>
+    )
+  }
+
+  wrapInItem(component) {
+    return <Item>{component}</Item>
+  }
+}
+
+Toolbar.propTypes = {
+  moreMenuTitle: PropTypes.string.isRequired,
+  moreMenuDisabled: PropTypes.bool,
+  moreMenuClassName: PropTypes.string,
+  moreMenuButton: PropTypes.element.isRequired,
+  children: PropTypes.node,
+  breakpoints: PropTypes.object
+}
+
+Toolbar.defaultProps = {
+  moreMenuDisabled: false,
+  moreMenuClassName: ''
+}
+
+export class ToolbarAction extends Component {
+  render() {
+    try {
+      const { children, visible, highlighted, isSecondaryAction } = this.props
+
+      if (!visible) return null
+
+      const child = React.cloneElement(React.Children.only(children), {
+        highlighted
+      })
+
+      if (isSecondaryAction) return child
+      else {
+        const hiddenStyle = highlighted
+          ? styles['u-hide--mob']
+          : styles['u-hide--desk']
+        return <span className={hiddenStyle}>{child}</span>
+      }
+    } catch (_) {
+      return null
+    }
+  }
+} // needs to extend Component to have `props.children`
```

Comment:
If a component is only a `class` with a `render` function, does it make sense to just use a functional component?

Reply:
I'll try that again, because yes, a functional component would make sense. But apparently [`props.children`](https://github.com/developit/preact-compat/issues/366#issuecomment-295822533) isn't always added by preact on functional components? It certainly wasn't there when I tested it, but I've changed a few things since.